app/src/main/java/com/example/weatherandroidapp/viewmodels/WeatherGridViewModel.kt

```diff
@@ -1,33 +1,27 @@
 package com.example.weatherandroidapp.viewmodels
 
-import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.LiveData
 import androidx.lifecycle.ViewModel
+import androidx.lifecycle.asLiveData
 import androidx.lifecycle.viewModelScope
 import com.example.domain.entities.ResponseWeatherInfo
-import com.example.domain.usecase.UseCaseInterfaceGetCity
-import com.example.domain.utils.ServiceResult
+import com.example.domain.usecase.WeatherUseCaseInterface
 import com.example.weatherandroidapp.contracts.WeatherGridContract
-import com.example.weatherandroidapp.utils.Data
-import com.example.weatherandroidapp.utils.Event
-import com.example.weatherandroidapp.utils.Status
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.launch
 import kotlinx.coroutines.withContext
 
 class WeatherGridViewModel(
-    private val useCaseWeatherCity: UseCaseInterfaceGetCity
+    private val useCaseWeatherCity: WeatherUseCaseInterface
 ) : ViewModel(), WeatherGridContract.ViewModel {
-    private val mutableGridState: MutableLiveData<Event<Data<ResponseWeatherInfo>>> by lazy { MutableLiveData() }
+    private val flowWeatherInfo: LiveData<ResponseWeatherInfo> = useCaseWeatherCity.observeWeather().asLiveData()
```

Comment:
here this copuld be called LiveData at this point

Reply:
Done